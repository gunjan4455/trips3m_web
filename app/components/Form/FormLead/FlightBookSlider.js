import React from 'react';
import { PropTypes } from 'prop-types';

import {
  Flights
} from 'helpers/Icon/Icon';

class FlightBooking extends React.Component {
  constructor(props) {
    super(props);
    this.checkInitialValues = this.checkInitialValues.bind(this);
    this.handleChange = this.handleChange.bind(this);
    this.isChecked = this.isChecked.bind(this);
    this.state = {
      selected: this.checkInitialValues()
    };
  }

  checkInitialValues() {
    try {
      if (this.props.formValues) {
        const flightAnswerAttributes = this.props.formValues.flight_answer_attributes;
        if (flightAnswerAttributes && flightAnswerAttributes.optionId) {
          return flightAnswerAttributes.optionId;
        }
        return null;
      }
      return null;
    } catch (error) {
      return null;
    }
  }

  isChecked(optionId) {
    try {
      let index = -1;
      if (this.props.input.value !== '') {
        index = [this.props.input.value].findIndex(obj =>
          obj.option_id.toString() === optionId.toString());
      }
      return (index !== -1);
    } catch (e) {
      return false;
    }
  }

  handleChange(event) {
    this.props.input.onChange({ option_id: event.target.value, customization_id: '23' });
  }

  render() {
    return (
      <div>
        <div className="flight-included col-xs-8 pr0 pl0">
          <p className="fw9 relative m0"><span className="flight-icon"><Flights /></span>Flights To Be Included?</p>
        </div>
        <div className="col-xs-4 pr0 pl0 text-right">
          <div className="col-xs-6 pl0 pr0">
            <input
              type="radio"
              className="flight-radio hide"
              name="cust_23"
              id="flight-included1"
              value="97"
              defaultChecked={this.state.selected === '97'}
              checked={this.isChecked(97)}
              onChange={this.handleChange}
            />
            <label htmlFor="flight-included1" className="cursorP pr8 pl8"><p className="m0 pfc4 fw4 f12">YES</p></label>
          </div>
          <div className="col-xs-6 pl0 pr0">
            <input
              type="radio"
              className="flight-radio hide"
              name="cust_23"
              id="flight-included2"
              value="98"
              defaultChecked={this.state.selected === '98'}
              checked={this.isChecked(98)}
              onChange={this.handleChange}
            />
            <label htmlFor="flight-included2" className="cursorP pr8 pl8"><p className="m0 pfc4 fw4 f12">NO</p></label>
          </div>
        </div>
      </div>
    );
  }
}

FlightBooking.propTypes = {
  input: PropTypes.object.isRequired,
  formValues: PropTypes.object
};

FlightBooking.defaultProps = {
  formValues: {}
};

export default FlightBooking;
