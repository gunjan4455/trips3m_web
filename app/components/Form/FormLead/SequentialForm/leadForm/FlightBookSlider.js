import React from 'react';
import { PropTypes } from 'prop-types';

import { FlightFilled, Tick } from 'helpers/Icon/Icon';

class FlightBooking extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: this.checkInitialValues(),
      flightSelected: '98',
    };
  }

  checkInitialValues = () => {
    try {
      if (this.props.formValues) {
        const flightAnswerAttributes = this.props.formValues.flight_answer_attributes;
        if (flightAnswerAttributes && flightAnswerAttributes.option_id) {
          return flightAnswerAttributes.option_id;
        }
        return null;
      }
      return null;
    } catch (error) {
      return null;
    }
  };

  handleChange = (event) => {
    if (!this.state.selected || this.state.selected === '98') {
      this.setState({ selected: '97' });
      this.props.input.onChange({
        option_id: '97',
        customization_id: '23',
      });
    } else {
      this.setState({ selected: '98' });
      this.props.input.onChange({
        option_id: '98',
        customization_id: '23',
      });
    }
  };

  render() {
    return (
      <div className={`button-box round-icon ${
          this.state.selected && this.state.selected !== '98' ? 'active' : ''
        }`}>
        <span className="tick-icon-container">
          <Tick/>
        </span>
        <div className="icon-48">
          <FlightFilled />
        </div>
        <input
          type="checkbox"
          className="checkbox-common"
          name="cust_23"
          id="flight-included1"
          value={this.state.selected}
          defaultChecked={this.state.selected === '97'}
          onChange={this.handleChange}
        />
        <p className="text-center">Flights</p>
      </div>
    );
  }
}

FlightBooking.propTypes = {
  input: PropTypes.object.isRequired,
  formValues: PropTypes.object
};

FlightBooking.defaultProps = {
  formValues: {}
};

export default FlightBooking;

