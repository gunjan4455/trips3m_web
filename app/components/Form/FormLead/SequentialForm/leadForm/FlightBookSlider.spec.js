import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import FlightBookSlider from './FlightBookSlider';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  input: {
    name: '"flight_answer_attributes"',
    value: '',
    onChange: () => {
    }
  },
  meta: {
    active: false,
    asyncValidating: false,
    autofilled: false,
    dirty: false,
    form: 'LeadForm',
    invalid: false,
    pristine: true,
    submitting: false,
    submitFailed: false,
    touched: false,
    valid: true,
    visited: false
  },
  formValues: {
    email: null,
    dont_utilize_budget_new: true,
    budget_new: true,
    start_date_noidea: false,
    adult: '2',
    child: 0,
    isd_code: '+91',
    child_age_answers_attributes: [
      {
        user_input: '0',
        customization_id: '7',
        option_id: '29'
      },
      {
        user_input: '0',
        customization_id: '7',
        option_id: '30'
      },
      {
        user_input: '0',
        customization_id: '7',
        option_id: '31'
      }
    ],
    syncErrors: {
      email: 'Enter valid email id'
    },
    to_loc: 'Kerala',
    start_date: '2019-07-17',
    start_date_flexible: false,
    trip_days: '2',
    hotel_needed_answer_attributes: {
      option_id: '95',
      customization_id: '22'
    },
    hotel_category_answers_attributes: [
      {
        option_id: '7',
        customization_id: '2'
      }
    ]
  }
};

const shallowWrapperFlightBookSlider = (props) => {
  return shallow(<FlightBookSlider {...props} />);
};

describe('lead form module', () => {

  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = shallowWrapperFlightBookSlider(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call handleChange properly',() => {
    const wrapper = shallowWrapperFlightBookSlider(props);
    wrapper.instance().handleChange({});
    expect(wrapper.state().selected).toEqual('97');
  });

  it('should call handleChange properly',() => {
    props.formValues.flight_answer_attributes = {
      customization_id: "23",
      option_id: "97"
    };
    const wrapper = shallowWrapperFlightBookSlider(props);
    wrapper.instance().handleChange({});
    expect(wrapper.state().selected).toEqual('98');
  });

});
