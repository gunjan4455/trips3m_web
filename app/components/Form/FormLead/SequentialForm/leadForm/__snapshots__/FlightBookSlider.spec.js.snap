// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lead form module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FlightBooking
    formValues={
      Object {
        "adult": "2",
        "budget_new": true,
        "child": 0,
        "child_age_answers_attributes": Array [
          Object {
            "customization_id": "7",
            "option_id": "29",
            "user_input": "0",
          },
          Object {
            "customization_id": "7",
            "option_id": "30",
            "user_input": "0",
          },
          Object {
            "customization_id": "7",
            "option_id": "31",
            "user_input": "0",
          },
        ],
        "dont_utilize_budget_new": true,
        "email": null,
        "hotel_category_answers_attributes": Array [
          Object {
            "customization_id": "2",
            "option_id": "7",
          },
        ],
        "hotel_needed_answer_attributes": Object {
          "customization_id": "22",
          "option_id": "95",
        },
        "isd_code": "+91",
        "start_date": "2019-07-17",
        "start_date_flexible": false,
        "start_date_noidea": false,
        "syncErrors": Object {
          "email": "Enter valid email id",
        },
        "to_loc": "Kerala",
        "trip_days": "2",
      }
    }
    input={
      Object {
        "name": "\\"flight_answer_attributes\\"",
        "onChange": [Function],
        "value": "",
      }
    }
    meta={
      Object {
        "active": false,
        "asyncValidating": false,
        "autofilled": false,
        "dirty": false,
        "form": "LeadForm",
        "invalid": false,
        "pristine": true,
        "submitFailed": false,
        "submitting": false,
        "touched": false,
        "valid": true,
        "visited": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="tick-icon-container"
        >
          <IconSvg />
        </span>,
        <div
          className="icon-48"
        >
          <IconSvg />
        </div>,
        <input
          className="checkbox-common"
          defaultChecked={false}
          id="flight-included1"
          name="cust_23"
          onChange={[Function]}
          type="checkbox"
          value={null}
        />,
        <p
          className="text-center"
        >
          Flights
        </p>,
      ],
      "className": "button-box round-icon ",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "tick-icon-container",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "icon-48",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "checkbox-common",
          "defaultChecked": false,
          "id": "flight-included1",
          "name": "cust_23",
          "onChange": [Function],
          "type": "checkbox",
          "value": null,
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Flights",
          "className": "text-center",
        },
        "ref": null,
        "rendered": "Flights",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="tick-icon-container"
          >
            <IconSvg />
          </span>,
          <div
            className="icon-48"
          >
            <IconSvg />
          </div>,
          <input
            className="checkbox-common"
            defaultChecked={false}
            id="flight-included1"
            name="cust_23"
            onChange={[Function]}
            type="checkbox"
            value={null}
          />,
          <p
            className="text-center"
          >
            Flights
          </p>,
        ],
        "className": "button-box round-icon ",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "tick-icon-container",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "icon-48",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "checkbox-common",
            "defaultChecked": false,
            "id": "flight-included1",
            "name": "cust_23",
            "onChange": [Function],
            "type": "checkbox",
            "value": null,
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Flights",
            "className": "text-center",
          },
          "ref": null,
          "rendered": "Flights",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
